kernel/src/kcb.rs

```diff
@@ -383,10 +396,42 @@ impl<A: ArchSpecificKcb> Kcb<A> {
         }
     }
 
+    pub fn set_pmem_affinity(&mut self, node: atopology::NodeId) -> Result<(), KError> {
```

Comment:
I also wonder can we make a generic set_allocation_affinity that gets the either pmem_arenas or memory_arenas as an argument and avoid code duplication here?

Reply:
The pattern seems weird; the outer layer will be sending the information to `set_allocation_affinity` that is already a part of KCB (arena and mem manager).

If you want to make one copy, maybe I can add a macro and change the variable names such that the shared code would be in the macro?

Reply:
Committed something let me know if this is bad if I missed something

Reply:
Ah, I "somehow" assumed that you want one `pub fn set_allocation_affinity` that others call. Hence the comment about passing arenas from outside.

This makes more sense, separate functions for Mem and PMem, and shared logic internally.